packages/fxa-auth-server/lib/payments/google-play/subscriptions.ts

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Container from 'typedi';
+import { internalValidationError } from '../../../lib/error';
+import { AppConfig } from '../../types';
+import { PlayBilling } from './play-billing';
+import { AbbrevPlayPurchase } from 'fxa-shared/subscriptions/types';
+import { SubscriptionPurchase } from './subscription-purchase';
+
+// TODO move this when we add support for Apple IAP subscriptions
+export interface SubscriptionsService<T> {
+  getSubscriptions: (uid: string) => Promise<T[]>;
+}
+
+/**
+ * Extract an AbbrevPlayPurchase from a SubscriptionPurchase
+ */
+export function abbrevPlayPurchaseFromSubscriptionPurchase(
+  purchase: SubscriptionPurchase
+): AbbrevPlayPurchase {
+  return {
+    auto_renewing: purchase.autoRenewing,
+    expiry_time_millis: purchase.expiryTimeMillis,
+    package_name: purchase.packageName,
+    sku: purchase.sku,
+    ...(purchase.cancelReason && { cancel_reason: purchase.cancelReason }),
+  };
+}
+
+export class PlaySubscriptions
+  implements SubscriptionsService<AbbrevPlayPurchase>
+{
+  private playBilling?: PlayBilling;
+
+  constructor() {
+    const config = Container.get(AppConfig);
+    if (!config.subscriptions.enabled) {
+      throw internalValidationError(
+        'PlaySubscriptions',
+        {},
+        new Error(
+          'Trying to new up PlaySubscriptions while subscriptions are disabled.  Check your dependency graph.'
+        )
+      );
+    }
+
+    if (Container.has(PlayBilling)) {
+      this.playBilling = Container.get(PlayBilling);
+    }
+  }
+
+  async getSubscriptions(uid: string) {
```

Comment:
This method name is a little misleading IMO as it returns `AbbrevPlayPurchase[]`, not `GooglePlaySubscription[]`.

I renamed this method in #10964 to `getAbbrevPlayPurchases` and added a new method to the new `playSubscriptions` class created in this PR called `getActiveGooglePlaySubscriptions` whose return type is `GooglePlaySubscription[]`. What do you think about these changes? I can handle any name changes in my PR if preferred.

Reply:
https://github.com/mozilla/fxa/pull/10948/files#r747067552 because I lost my ability to use Github?